### PR TITLE
#5049: ensure YAML bricks are shown in add brick modal

### PR DESCRIPTION
**Pull request by twschiller**
## What does this PR do?

- Fixes: #5049 
- There was a bug in the registry fetch method where it would remove the items from the cache and save them to IDB, but not re-register them
- Renames useRefresh to useRefreshRegistries

## Checklist

- 😞  Add tests: we don't have unit tests for baseRegistry yet. I'll add some after we cut a beta with this PR
- [x] Designate a primary reviewer: @BALEHOK

### CODE REVIEW

File: src/baseRegistry.ts [hunk 1]
```diff
@@ -239,19 +257,31 @@ export class Registry<
...
-    this.notifyAll();
+    // Mark as being from the remote server
+    for (const item of packages) {
+      this.remote.add(item.id as RegistryId);
+    }
+
+    // Force reload of all items from IDB. To avoid hitting IDB, we could just re-register the items that were retrieved
+    // locally. However, the idea of syncRemote is that it might also remove bricks that are no longer
+    // available/accessible to the user.
+    await this.all();
```
Comment (twschiller) [inline src/baseRegistry.ts:284]: This is the primary functional fix

File: src/baseRegistry.ts [hunk 2]
```diff
@@ -239,19 +257,31 @@ export class Registry<
...
+            kind
+          );
           await registry.syncRemote(kind as Kind, kindPackages);
         }
       )
     );
 
-    this.notifyAll();
+    // Mark as being from the remote server
+    for (const item of packages) {
```
Comment (twschiller) [inline src/baseRegistry.ts:277]: Waiting for the bricks to be persisted in IDB is a bit more accurate on when to update this.remote

We don't really use this.remote for anything functional, it's just bookkeeping

File: src/baseRegistry.ts [hunk 3]
```diff
@@ -228,8 +248,6 @@ export class Registry<
         );
       }
 
-      this.cache.delete(item.metadata.id);
```
Comment (twschiller) [inline src/baseRegistry.ts:231]: This was the call that was clearing out elements from the cache that were already hydrated from IDB

File: src/baseRegistry.ts [hunk 4]
```diff
@@ -52,6 +56,10 @@ export class Registry<
 
   private readonly cache = new Map<RegistryId, Item>();
 
+  /**
+   * Set of ids that have been retrieved from server.
+   * @private
```
Comment (BLoe) [inline src/baseRegistry.ts:61]: What does the `@private` jsdoc tag here do/mean? Just indicating that the field is private in the jsdoc popups?

**Review by BLoe — APPROVED** [review]
LGTM 🚀

**Pull request merged**